File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/LaserAim.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class LaserAim extends Equipable {
	{
		name = Locale.get("laser_aim");
		description = Locale.get("laser_aim");
		sprite = "item-laser_pointer";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).hasRedLine = true;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).hasRedLine = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/LaserAim.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class LaserAim : Equippable() {
	init {
		name = Locale.get("laser_aim")
		description = Locale.get("laser_aim")
		sprite = "item-laser_pointer"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.hasRedLine = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.hasRedLine = false
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun canBeDegraded(): Boolean {
		return false
	}
}